Stopping after X revs

tomasmckenna opened the pull request:
The digital crown sets the revolutions before it has started (after that, it sets the speed). Todo is to fix the animation for when isFinished is true (in ContentView).

Also, why do I have to set the text Text("") for it to work?

Plus, is there any less ugly ways then having an if in the update function for elapsedTime?

tomasmckenna committed:
Stopping after X revs

The digital crown sets the revolutions before it has started (after that, it sets the speed). Todo is to fix the animation for when isFinished is true (in ContentView).

Also, why do I have to set the text Text("") for it to work?

Plus, is there any less ugly ways then having an if in the update function for elapsedTime?

tomasmckenna committed:
As far as i can get with drag gesture

I have added stuff from a tutorial (https://www.hackingwithswift.com/books/ios-swiftui/moving-views-with-draggesture-and-offset) but now I cannot figure out how to load the InstructionView() since it is protesting that the result is not used.

Also, sorry for adding it to the same branch as the other fix. Too many merge conflicts in my new branch to solve.

blomma commented on HertzWatch WatchKit Extension/ContentView.swift:
```diff
@@ -146,6 +136,18 @@ struct ContentView: View {
+                            }
+                            .onEnded { gesture in
+                                if abs(self.offset.height) > 20 {
+                                    InstructionView()
```
So here's your issue, you got bugs in the carbonator. As you noticed, calling a InstructionView from here won't work for the simple reason that it's a view. What you need to do is add another @State variable and then in the main view code, that is in the same way we check for example if the model.isRunning you would do the same thing with the new State variable.

tomasmckenna committed:
Success icon when finished session

Still needs to fix these things to be entirely done (and I don't know how to fix them):
* Create an array with all of the success images
* Create a function that shows a success image based on the max diff between high (breathIn) and low (breathOut) heartrate value and prints out the max diff as well (today it just says "20").
* Later when we have added graphs, we could add a support text underneeth the success icon/text that says "See your graph when you swipe down"

tomasmckenna committed:
Fixed instructions (draft 1 though) and success stats

Success stats tells you the score - right now printing out two values but in the future, Tomás will pick one of them so don't mind that it looks a bit crammed with space.

Added instructions but some tweaking might be needed.

tomasmckenna committed:
updated instrucitons a little. Would like to left align them, and maybe add a graphic

Also I am not sur ebut I think it has the same limit we had before, if I add too much stuff it doesn't work, so I guess I need to group them if we want to have long, scroilable content.

blomma committed:
Sorted out version bumpelibump
